adapters/repos/db/shard.go

```diff
@@ -32,21 +37,129 @@ import (
 	"github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw"
 	"github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw/distancer"
 	"github.com/weaviate/weaviate/adapters/repos/db/vector/noop"
+	"github.com/weaviate/weaviate/entities/additional"
+	"github.com/weaviate/weaviate/entities/aggregation"
+	"github.com/weaviate/weaviate/entities/backup"
 	"github.com/weaviate/weaviate/entities/cyclemanager"
 	"github.com/weaviate/weaviate/entities/filters"
 	"github.com/weaviate/weaviate/entities/models"
 	"github.com/weaviate/weaviate/entities/schema"
+	"github.com/weaviate/weaviate/entities/search"
 	"github.com/weaviate/weaviate/entities/storagestate"
 	"github.com/weaviate/weaviate/entities/vectorindex"
 	"github.com/weaviate/weaviate/entities/vectorindex/common"
 	flatent "github.com/weaviate/weaviate/entities/vectorindex/flat"
 	hnswent "github.com/weaviate/weaviate/entities/vectorindex/hnsw"
 	"github.com/weaviate/weaviate/usecases/monitoring"
+	"github.com/weaviate/weaviate/usecases/replica"
 	"golang.org/x/sync/errgroup"
+
+	"github.com/weaviate/weaviate/entities/storobj"
+
+	"github.com/weaviate/weaviate/usecases/objects"
 )
 
 const IdLockPoolSize = 128
 
+type ShardLike interface {
```

Comment:
Some of this interface's methods are commented out, @aliszka what is the plan for these?

Reply:
They can be removed. I was commenting them instead of deleting just in case they will turn out needed eventually. Apparently they are shard's internals and do not have to be exposed in the interface